tests/expressions_window/tests.py

```diff
@@ -0,0 +1,880 @@
+import datetime
+from unittest import skipIf, skipUnless
+
+from django.core.exceptions import FieldError
+from django.db import NotSupportedError, connection
+from django.db.models import F, RowRange, ValueRange, Window
+from django.db.models.aggregates import Avg, Max, Min, Sum
+from django.db.models.expressions import Value
+from django.db.models.functions import (
+    CumeDist, DenseRank, ExtractYear, FirstValue, Lag, LastValue, Lead,
+    NthValue, Ntile, PercentRank, Rank, RowNumber, Upper,
+)
+from django.test import TestCase, skipUnlessDBFeature
+
+from .models import WindowTestModel
+
+
+@skipUnlessDBFeature('supports_over_clause')
+class TestWindowFunction(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        """
+        The test data was designed to accommodate the requirements to support
+        all of the different types of window functions as well as partitioning
+        and frames.
+
+        The test data was partially inspired by the documentation in
+        PostgreSQL, and from some third-party material that explained
+        window functions.
+        """
+        WindowTestModel.objects.create(
```

Comment:
Since you don't actually need the instances of all those models, can you use `WindowTestModel.objects.bulk_create()` please.